tests/unit/test_tiler.py

```diff
@@ -0,0 +1,61 @@
+import os
+
+import pytest
+
+from src.histolab.slide import Slide
+from src.histolab.tiler import RandomTiler, Tiler
+
+from ..unitutil import ANY, PILImageMock, class_mock, initializer_mock
+
+
+class Describe_RandomTiler(object):
+    def it_constructs_from_args(self, request):
+        _init = initializer_mock(request, RandomTiler)
+
+        random_tiler = RandomTiler((512, 512), 10, 2, 7, True, "", ".png", 1e4)
+
+        _init.assert_called_once_with(ANY, (512, 512), 10, 2, 7, True, "", ".png", 1e4)
+        assert isinstance(random_tiler, RandomTiler)
+        assert isinstance(random_tiler, Tiler)
+
+    def but_it_has_wrong_tile_size_value(self, request):
+        with pytest.raises(ValueError) as err:
+            RandomTiler((512, -1), 10, 0)
+
+        assert isinstance(err.value, ValueError)
+        assert str(err.value) == "Tile size cannot be negative ((512, -1))"
+
+    def or_it_has_not_available_level_value(self, request, tmpdir):
+        tmp_path_ = tmpdir.mkdir("myslide")
+        image = PILImageMock.DIMS_50X50_RGB_RANDOM_COLOR
+        image.save(os.path.join(tmp_path_, "mywsi.png"), "PNG")
+        slide_path = os.path.join(tmp_path_, "mywsi.png")
+        slide = Slide(slide_path, "processed")
+        random_tiler = RandomTiler((512, 512), 10, 3)
+
+        with pytest.raises(IndexError) as err:
+            random_tiler.extract(slide)
+
+        assert isinstance(err.value, IndexError)
```

Comment:
I'm not sure this is a significative test, for sure here we're simulating a wrong user behaviour ... but in future these kind of exceptions could be managed as custom, so we will have "our" exception rather than checking for the builtin exception. NOTE: this kind of error returned to a standard user is not meaningful

Reply:
`or_it_has_not_available_level_value`

Reply:
Yes, we are simulating a wrong user behavior that could totally happen.

To have a more meaningful message I could catch the exception in `level_dimensions()` and raise a `ValueError` instead - later to be turned into our custom exception. What do you think?

Reply:
i do agree let's do this thing :D